Sample/app/src/main/java/com/microsoft/projectoxford/visionsample/HandwritingRecognizeActivity.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_recognize, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    // Called when the "Select Image" button is clicked.
+    public void selectImage(View view) {
+        mEditText.setText("");
+
+        Intent intent;
+        intent = new Intent(HandwritingRecognizeActivity.this, com.microsoft.projectoxford.visionsample.helper.SelectImageActivity.class);
+        startActivityForResult(intent, REQUEST_SELECT_IMAGE);
+    }
+
+    // Called when image selection is done.
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Log.d("AnalyzeActivity", "onActivityResult");
+        switch (requestCode) {
+            case REQUEST_SELECT_IMAGE:
+                if (resultCode == RESULT_OK) {
+                    // If image is selected successfully, set the image URI and bitmap.
+                    mImageUri = data.getData();
+
+                    mBitmap = ImageHelper.loadSizeLimitedBitmapFromUri(
+                            mImageUri, getContentResolver());
+                    if (mBitmap != null) {
+                        // Show the image on screen.
+                        ImageView imageView = (ImageView) findViewById(R.id.selectedImage);
+                        imageView.setImageBitmap(mBitmap);
+
+                        // Add detection log.
+                        Log.d("AnalyzeActivity", "Image: " + mImageUri + " resized to " + mBitmap.getWidth()
+                                + "x" + mBitmap.getHeight());
+
+                        doRecognize();
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+
+    public void doRecognize() {
+        mButtonSelectImage.setEnabled(false);
+        mEditText.setText("Analyzing...");
+
+        try {
+            new doRequest().execute();
+        } catch (Exception e) {
+            mEditText.setText("Error encountered. Exception is: " + e.toString());
+        }
+    }
+
+    private String process() throws VisionServiceException, IOException, InterruptedException {
+        Gson gson = new Gson();
+
+        // Put the image into an input stream for detection.
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
```

Comment:
Although it has no effect without closing the ByteArrayOutputStream&ByteArrayInputStream , it's always a good practice to close them.